client/emv/cmdemv.c

```diff
@@ -0,0 +1,537 @@
+				case 'K':
+					leaveSignalON = true;
+					break;
+				case 't':
+				case 'T':
+					decodeTLV = true;
+					break;
+				default:
+					PrintAndLog("Unknown parameter '%c'", param_getchar_indx(cmd, 1, cmdp));
+					return 1;
+		}
+		cmdp++;
+	}
+
+	struct tlvdb *t = NULL;
+	const char *al = "Applets list";
+	t = tlvdb_fixed(1, strlen(al), (const unsigned char *)al);
+
+	if (EMVSearch(activateField, leaveSignalON, decodeTLV, t)) {
+		tlvdb_free(t);
+		return 2;
+	}
+	
+	PrintAndLog("Search completed.");
+
+	// print list here
+	if (!decodeTLV) {  
+		TLVPrintAIDlistFromSelectTLV(t);
+	}
+	
+	tlvdb_free(t);
+	
+	return 0;
+}
+
+int UsageCmdHFEMVPPSE(void) {
+	PrintAndLog("HELP :  Executes PSE/PPSE select command. It returns list of applet on the card:\n");
+	PrintAndLog("Usage:  hf emv pse [-s][-k][-1][-2][-t]\n");
+	PrintAndLog("  Options:");
+	PrintAndLog("  -s       : select card");
+	PrintAndLog("  -k       : keep field for next command");
+	PrintAndLog("  -1       : ppse (1PAY.SYS.DDF01)");
+	PrintAndLog("  -2       : pse (2PAY.SYS.DDF01)");
+	PrintAndLog("  -t       : TLV decode results\n");
+	PrintAndLog("Samples:");
+	PrintAndLog(" hf emv pse -s 1 -> select, get pse");
+	PrintAndLog(" hf emv pse -s -k 2 -> select, get ppse, keep field");
+	PrintAndLog(" hf emv pse -s -t 2 -> select, get ppse, show result in TLV");
+	return 0;
+}
+
+int CmdHFEMVPPSE(const char *cmd) {
+	
+	uint8_t PSENum = 2;
+	bool activateField = false;
+	bool leaveSignalON = false;
+	bool decodeTLV = false;
+
+	if (strlen(cmd) < 1) {
+		UsageCmdHFEMVPPSE();
+		return 0;
+	}
+
+	SetAPDULogging(true);
+	
+	int cmdp = 0;
+	while(param_getchar(cmd, cmdp) != 0x00) {
+		char c = param_getchar(cmd, cmdp);
+		if ((c == '-') && (param_getlength(cmd, cmdp) == 2))
+			switch (param_getchar_indx(cmd, 1, cmdp)) {
+				case 'h':
+				case 'H':
+					UsageCmdHFEMVPPSE();
+					return 0;
+				case 's':
+				case 'S':
+					activateField = true;
+					break;
+				case 'k':
+				case 'K':
+					leaveSignalON = true;
+					break;
+				case 't':
+				case 'T':
+					decodeTLV = true;
+					break;
+				case '1':
+					PSENum = 1;
+					break;
+				case '2':
+					PSENum = 2;
+					break;
+				default:
+					PrintAndLog("Unknown parameter '%c'", param_getchar_indx(cmd, 1, cmdp));
+					return 1;
+		}
+		cmdp++;
+	}
+	
+	// exec
+	uint8_t buf[200] = {0};
+	size_t len = 0;
+	uint16_t sw = 0;
+	int res = EMVSelectPSE(activateField, leaveSignalON, PSENum, buf, sizeof(buf), &len, &sw);
+	
+	if (sw)
+		PrintAndLog("APDU response status: %04x - %s", sw, GetAPDUCodeDescription(sw >> 8, sw & 0xff)); 
+
+	if (res)
+		return res;
+	
+	
+	if (decodeTLV)
+		TLVPrintFromBuffer(buf, len);
+
+	return 0;
+}
+
+int UsageCmdHFEMVExec(void) {
+	PrintAndLog("HELP :  Executes EMV contactless transaction:\n");
+	PrintAndLog("Usage:  hf emv exec [-s][-a][-t]\n");
+	PrintAndLog("  Options:");
+	PrintAndLog("  -s       : select card");
+	PrintAndLog("  -a       : show APDU reqests and responses\n");
```

Comment:
Why do we have a -a option here and not in other emv commands?

Reply:
Ill add it

Reply:
or.......   the rest commands shows apdu by default. 
so add this option?

Reply:
I can add this option only to search. search produces many lines to console...

Reply:
For the sake of consistency I propose to have it for all emv commands.